src/concat_source.rs

```diff
@@ -61,17 +61,17 @@ impl ConcatSource {
   /// Create a [ConcatSource] with [Source]s.
   pub fn new<S, T>(sources: S) -> Self
   where
-    T: Into<BoxSource>,
+    T: Source + 'static,
```

Comment:
This part doesn't seem to make any difference to me. Could you elaborate more?

Reply:
Is it necessary for us to add `impl Source for Box<dyn Source>` or maybe there are some cases that I missed? From my perspective of view, it looks like `Box<dyn Source>` is enough

Reply:
Something like this seems enough to me: `ReplaceSource::new(RawSource::from("raw"));`

Reply:
For example, `Module` in rspack will have a `originalSource` like webpack, it's created by loaders result, it might be an `SourceMapSouce` (when source map enabled) or a `RawSource` (when source map disabled), so we can't know the what exact type it is, and the `ReplaceSource<Box<dyn Source>>` situation will happen when we need to replace/modify something on the `Module.originalSource`.

Reply:
You are right, LGTM